### ISSUE TITLE
Issue running get_my_data

### ISSUE THREAD

**Issue by shromada** [cellular-tracking-technologies/celltracktech#13]
I am unable to run the get_my_data() function. This was previously not an issue before the last package update, but now the function fails with the following error when ran with the prior : 
`Error in FUN(X[[i]], ...) : 
  argument "filetypes" is missing, with no `default``
  This error returns after the function calculates how many files will be downloaded (8883) and prints:
 `[1] "prepped list of filenames to get"`. 
I tried to specify the `filetypes = c("raw", "node_health", "gps", "blu")` in the function call using what seems to be expected in downstream functions, but that results in the same error. 

I am using R version 4.4.1, but had the same issue with an install of R 4.2.1. The error is returned whether or not I have actually connected to the postgres database.

**Comment by Lisark45** [comment]
I am also getting this error message. Has there been any resolution please?

**Comment by shromada** [comment]
There has been no resolution as far as this package. I have been able to successfully run the same function in CTT's other repo: https://github.com/cellular-tracking-technologies/data_tools

**Comment by Lisark45** [comment]
Thank you so much. Worked for me too :)

**Comment by meelyn-pandit** [comment]
Hi @shromada and @Lisark45,

I am sorry that this issue was not resolved sooner. I am a developer at CTT working on updating this package. Could you attach the R scripts you used that resulted in this error?

**Comment by shromada** [comment]
It seems that we cannot upload .R files. This is the code:

`my_token <- "42ec53e7390c14801f8edf8e3fc30523db2068b472d71d4fe3d7563cd7c18436"
db_name <- "postgres"
myproject <- "Lizards on the Panoche Plateau" #this is your project name on your CTT account
#conn <- dbConnect(RPostgres::Postgres(), dbname=db_name,user = 'postgres', password = 'database')

conn <- dbConnect(RPostgres::Postgres(), 
                 user = "X", 
                 password = "XX", 
                 dbname = "fcz_conservation_db",
                 host = "fcz-conservation.cxsi60ieiqbe.us-west-1.rds.amazonaws.com",
                 port = "5432",
                 options="-c search_path=public")

################

#this will pull all new data from the sensor stations
get_my_data(my_token, "./mydata/" ,db_name = conn, myproject = myproject, 
            begin = "2024-6-01 00:00:00 UTC")#, filetypes = NULL) #the folder path is where you want your downloaded files to go
`

**Comment by meelyn-pandit** [comment]
Hi @shromada,

I am running this script now and I am able to download data. Once it is done I will send you the script I have.

From first glance, your 'db_name' variable needs to have '.db' at the end of it; so replace it with:

`db_name <- 'postgres.db'`

**Comment by meelyn-pandit** [comment]
Hi @shromada,

Here is the script I used to download your data. I used duckdb since I already have that installed on my computer. You were using RPostgres to connect to an existing database on AWS, correct?

If you still have issues, create a customer service desk ticket here at the link below. That way more developers can be aware of the issue.

https://celltracktech.com/pages/customer-service-desk-csd

`
install.packages("duckdb", repos = c("https://duckdb.r-universe.dev", "https://cloud.r-project.org"))
library(duckdb)
library(celltracktech)
library(DBI)

my_token <- "42ec53e7390c14801f8edf8e3fc30523db2068b472d71d4fe3d7563cd7c18436"
db_name <- "postgres.db"
myproject <- "Lizards on the Panoche Plateau" #this is your project name on your CTT account

# connect to database using RPostgres
#conn <- dbConnect(RPostgres::Postgres(), dbname=db_name,user = 'postgres', password = 'database')
#
# conn <- dbConnect(RPostgres::Postgres(),
#                   user = "X",
#                   password = "XX",
#                   dbname = "fcz_conservation_db",
#                   host = "fcz-conservation.cxsi60ieiqbe.us-west-1.rds.amazonaws.com",
#                   port = "5432",
#                   options="-c search_path=public")

# connect to database using duckdb
conn <- DBI::dbConnect(
  duckdb::duckdb(),
  dbdir = "./mydata/postgres.db",
  read_only = FALSE
)
################

#this will pull all new data from the sensor stations
get_my_data(my_token,
            "./mydata/",
            db_name = conn,
            myproject = myproject,
            begin = "2024-6-01 00:00:00 UTC",
            end = "2024-06-02",
            filetypes = c('raw', 'node-health'))#, filetypes = NULL) #the folder path is where you want your downloaded files to go
dbDisconnect(conn)
`

**Comment by shromada** [comment]
@meelyn-pandit 

Yes, I had been using RPostgres to connect to our AWS hosted database. I am not very well versed with the options here, what is the syntax to use to connect to the AWS server? 

When I adjust your code to remove the end date in the get_my_data, it successfully downloads the data to my local storage in my R project but does not push to the AWS hosted database.

**Comment by meelyn-pandit** [comment]
@shromada 

I have not connected to an AWS database before but I am looking into it.

In the meantime I would continue to use RPostgres instead since that has support for AWS RDS databases.

Out of curiosity, did you manually create the 'mydata' folder? I was getting errors when I forgot to manually create that folder in my working directory.

**Comment by shromada** [comment]
I do not remember if I had manually created the 'mydata' folder, but the function in the package was able to drop data into it so I do not think that is the issue. 

Again, the version of the get_my_data() function in this [repo](https://github.com/cellular-tracking-technologies/data_tools) still works, just not the version in this package.

Maybe this issue has something to do with database structure? I remember seeing a post on the CTT slack encouraging rebuilding the databases. I was going to do that, but since this new version of the function did not work, I held off. I can no longer see the discussion, as the post on slack was too long ago to be viewable with the free slack level.

Would it be best for me to start another support ticket, or should I add it to my current one about tower checkins?

Thanks!

**Comment by meelyn-pandit** [comment]
Hi Steve,

Yes please create a new support ticket. I’ll check the data tools repo and
see what the the differences are.

Best,

Meelyn

**Comment by meelyn-pandit** [comment]
Latest package update fixes issue, you will need to install the package in R again to upload data to a remote database.

`library(devtools)
install_github('cellular-tracking-technologies/celltracktech')`
